cloud-sql/mysql/tink/main.py

```
@@ -0,0 +1,237 @@
+            time_cast = row[1]
+
+            # Print recent votes
+            print("{}\t{}\t{}".format(team, email, time_cast))
+
+        stmt = sqlalchemy.text(
+            "SELECT COUNT(vote_id) FROM votes WHERE candidate=:candidate"
+        )
+        # Count number of votes for tabs
+        tab_result = conn.execute(stmt, candidate="TABS").fetchone()
+        tab_count = tab_result[0]
+        # Count number of votes for spaces
+        space_result = conn.execute(stmt, candidate="SPACES").fetchone()
+        space_count = space_result[0]
+
+        # Print total votes
+        print("")
+        print("Total: {} spaces, {} tabs".format(space_count, tab_count))
+# [END cloud_sql_mysql_cse_query]
+
+
+# [START cloud_sql_mysql_cse_insert]
+def add_vote(team, email, db, eaead):
+    import sqlalchemy
+
+    time_cast = datetime.datetime.utcnow()
+    # Encrypt the email of the voter
+    encrypted_email = eaead.encrypt(email.encode(), team.encode())
+    # Verify that the team is one of the allowed options
+    if team != "TABS" and team != "SPACES":
+        logger.error("Invalid team specified: {}".format(team))
+        return
+
+    # Preparing a statement before hand can help protect against injections.
+    stmt = sqlalchemy.text(
+        "INSERT INTO votes (time_cast, candidate, voter_email)"
+        " VALUES (:time_cast, :candidate, :voter_email)"
+    )
+    try:
+        # Using a with statement ensures that the connection is always released
+        # back into the pool at the end of statement (even if an error occurs)
+        with db.connect() as conn:
+            conn.execute(stmt, time_cast=time_cast, candidate=team,
+                         voter_email=encrypted_email)
+    except Exception as e:
+        # If something goes wrong, handle the error in this section. This might
+        # involve retrying or adjusting parameters depending on the situation.
+        logger.exception(e)
```

Same here - should probably just let this bubble up.

---

Done